File: app/src/main/java/com/pavelprymak/propodcast/services/PlayerUI.java
package com.pavelprymak.propodcast.services;

public interface PlayerUI {
    // User Actions
    void playAction();

    void pauseAction();

    void stopAction();

    void startTrackAction(String trackLink, String trackTitle, String imageUrl,String trackAuthor);

    void updateUiAction();

    void seekToPositionAction(float percentSeekPosition);


    // UI preparing
    void setPlayStatus(boolean isPlay);

    void setLoadingStatus(boolean isLoading);

    void setPlayerErrors(String errorMessage);

    void setPlaybackDuration(long duration);

    void setPlaybackCurrentPosition(long currentPosition, long duration);

    void setTrackTitle(String trackTitle);

    void setTrackImage(String imageUrl);
}

File: app/src/main/java/com/pavelprymak/propodcast/services/PlayerUI.kt
package com.pavelprymak.propodcast.services

interface PlayerUI {
    // User Actions
    fun playAction()

    fun pauseAction()

    fun stopAction()

    fun startTrackAction(trackLink: String?, trackTitle: String?, imageUrl: String?, trackAuthor: String?)

    fun updateUiAction()

    fun seekToPositionAction(percentSeekPosition: Float)


    // UI preparing
    fun setPlayStatus(isPlay: Boolean)

    fun setLoadingStatus(isLoading: Boolean)

    fun setPlayerErrors(errorMessage: String)

    fun setPlaybackDuration(duration: Long)

    fun setPlaybackCurrentPosition(currentPosition: Long, duration: Long)

    fun setTrackTitle(trackTitle: String?)

    fun setTrackImage(imageUrl: String?)
}
